src/main/scala/com/microsoft/hyperspace/index/IndexCacheFactory.scala

```diff
@@ -25,14 +26,20 @@ object IndexCacheType {
 }
 
 trait IndexCacheFactory {
-  def create(spark: SparkSession, cacheType: String): Cache[Seq[IndexLogEntry]]
+  def create(spark: SparkSession, cacheType: String, clock: Clock = new SystemClock)
+    : Cache[Seq[IndexLogEntry]]
 }
 
 object IndexCacheFactoryImpl extends IndexCacheFactory {
-  override def create(spark: SparkSession, cacheType: String): Cache[Seq[IndexLogEntry]] = {
-    cacheType match {
-      case IndexCacheType.CREATION_TIME_BASED => new CreationTimeBasedIndexCache(spark)
-      case _ => throw HyperspaceException(s"Unknown cache type: $cacheType.")
-    }
+  override def create(spark: SparkSession, cacheType: String, clock: Clock = new SystemClock)
```

Comment:
Can we also remove default from here and pass `SystemClock` from `CachingIndexCollectionManager` so that it's easier to track which one is providing the specific impl.